src/Response/Iterator/MultiBulk.php

```diff
@@ -40,6 +40,8 @@ public function __construct(NodeConnectionInterface $connection, $size)
      */
     public function __destruct()
     {
+        // TODO: Disconnect on every iterator makes it fail during handshake
+        // TODO: (e.g connect() iterates over commands and eventually disconnects which kill the resource and makes connection broken)
```

Comment:
Do you want to resolve this before merging?

Reply:
Is it possible to postpone it a bit and remove from the scope of this PR? Have other priorities in meantime

Reply:
@tillkruss My bad, it wasn't a bug it was just an issue with test setup, now Iterator tests are executing correctly against protected server